statsmodels/base/tests/test_screening.py

```diff
@@ -0,0 +1,359 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed May 23 12:53:27 2018
+
+Author: Josef Perktold
+
+"""
+
+import numpy as np
+from numpy.testing import assert_allclose, assert_equal
+import pandas as pd
+
+from statsmodels.discrete.discrete_model import Poisson, Logit
+from statsmodels.genmod.generalized_linear_model import GLM
+from statsmodels.genmod.families import family
+from statsmodels.base._penalized import PenalizedMixin
+from statsmodels.base._screening import VariableScreening
+
+
+class PoissonPenalized(PenalizedMixin, Poisson):
+    pass
+
+
+class LogitPenalized(PenalizedMixin, Logit):
+    pass
+
+
+class GLMPenalized(PenalizedMixin, GLM):
+    pass
+
+
+def _get_poisson_data():
+    np.random.seed(987865)
+
+    nobs, k_vars = 100, 500
+    k_nonzero = 5
+    x = (np.random.rand(nobs, k_vars) +
+         1. * (np.random.rand(nobs, 1) - 0.5)) * 2 - 1
+    x *= 1.2
+
+    x = (x - x.mean(0)) / x.std(0)
+    x[:, 0] = 1
+    beta = np.zeros(k_vars)
+    idx_nonzero_true = [0, 100, 300, 400, 411]
+    beta[idx_nonzero_true] = 1. / np.arange(1, k_nonzero + 1)
+    beta = np.sqrt(beta)  # make small coefficients larger
+    linpred = x.dot(beta)
+    mu = np.exp(linpred)
+    y = np.random.poisson(mu)
+    return y, x, idx_nonzero_true, beta
+
+def test_poisson_screening():
+    # this is mostly a dump of my development notebook
+    # number of exog candidates is reduced to 500 to reduce time
+    np.random.seed(987865)
+
+    y, x, idx_nonzero_true, beta = _get_poisson_data()
+    nobs = len(y)
+
+    xnames_true = ['var%4d' % ii for ii in idx_nonzero_true]
+    xnames_true[0] = 'const'
+    parameters = pd.DataFrame(beta[idx_nonzero_true], index=xnames_true,
+                              columns=['true'])
+
+    xframe_true = pd.DataFrame(x[:, idx_nonzero_true], columns=xnames_true)
+    res_oracle = Poisson(y, xframe_true).fit()
+    parameters['oracle'] = res_oracle.params
+
+    mod_initial = PoissonPenalized(y, np.ones(nobs), pen_weight=nobs * 5)
+
+    screener = VariableScreening(mod_initial)
+    exog_candidates = x[:, 1:]
+    res_screen = screener.screen_exog(exog_candidates, maxiter=10)
+
+    res_screen.idx_nonzero
+
+    res_screen.results_final
```

Comment:
expression that don't have an effect
unit tests still need review for left overs from my initial script
more assert?  selection and params is asserted